src/helpers/folders.ts

```diff
@@ -63,8 +63,12 @@ export const getFolderOtherOwnerAccountName = (
 		return null;
 	}
 
-	// If the id contains the zid, the account is considered the owner if the zid matches the account id
-	const matchingFolderRoot = find(folderRoots, { zid });
+	/** find the folderRoots for which the id corresponds to the message zid
+	 * if the folderRoots has an owner, return the owner
+	 * if not, return null
+	 * */
+
+	const matchingFolderRoot = find(folderRoots, (c) => c.id.includes(zid));
```

Comment:
The _include_ comparison is kind of weak but it works because of the current format logic of the folder id and that's ok.
But as a possible future optimization I would suggest to use the _getFolderIdParts_ helper function or create a shortcut function for that, only for the zid (e.g. _getFolderZid_).
This way we don't spread the id logic all over the codebase, we can just change one function if there will be some evolution on the id format, and the code will be more self-describing because the function name will give more semantic context